pkg/controller/worker/machines.go

```diff
@@ -185,7 +191,11 @@ func (w *workerDelegate) generateMachineConfig() error {
 				Taints:               pool.Taints,
 				MachineConfiguration: genericworkeractuator.ReadMachineConfiguration(pool),
 			})
-
+			// TODO: remove when AWS CSI driver stops using the aws csi topology key - https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/899
+			// add aws csi driver topology label if its not specified
+			if csiEnabled {
+				machineDeployments[len(machineDeployments)-1].Labels = utils.MergeStringMaps(pool.Labels, map[string]string{awsCSIDriverTopologyKey: zone})
```

Comment:
I do not like very much this line. 
You can firstly populate a machine deployment object and then append it to the list of all machine deployments. 
Other option is to have a helper (lambda) function that calculate the labels and instead of 
```golang
Labels: pool.Labels,
```

you can have something like 
```golang
Labels: func() map[string]string {
	if !csiEnabled {
		return pool.Labels
	}
	return utils.MergeStringMaps(pool.Labels, map[string]string{awsCSIDriverTopologyKey: zone})
}(),

```

Reply:
find the changes here df1b320